caracal/workers/prep_worker.py

```diff
@@ -2,29 +2,110 @@
+    """
+    Shameless copy of the hetfield function to return field coordinates 
+    from the database.
+    Find match of fields in info
+    Parameters:
+    info (dict): dictionary of obsinfo as read by yaml
+    field (str): field name
+    db (dict):   calibrator data base as returned by
+                 calibrator_database()
+    Go through all calibrators in db and return the first that matches
+    the coordinates of field in msinfo. Return empty string if not
+    found.
+    If coordinates difference is larger than tol_diff, return the correct coordinates, else return empty string.
+    """
+
+    # Get position of field in msinfo
+    ind = info['FIELD']['NAME'].index(field)
+    firade = info['FIELD']['DELAY_DIR'][ind][0]
+    firade[0] = np.mod(firade[0],2*np.pi)
+    dbcp = db.db
+    for key in dbcp.keys():
+        carade = [dbcp[key]['ra'],dbcp[key]['decl']]
+        if closeby(carade, firade, tol=tol):
+            print("Closeby source found")
+            if not closeby(carade, firade, tol=tol_diff):
+               return key, dbcp[key]['ra'], dbcp[key]['decl']
+            else :
+               return key, '', ''
+    return False
+
+
 def worker(pipeline, recipe, config):
     label = config['label_in']
     wname = pipeline.CURRENT_WORKER
     field_name = config["field"]
-
     msdir = pipeline.msdir
     for i in range(pipeline.nobs):
         prefix_msbase = pipeline.prefix_msbases[i]
         mslist  = pipeline.get_mslist(i, label, target=(field_name == "target"))
 
+
         for msname in mslist:
             if not os.path.exists(os.path.join(msdir, msname)):
                 caracal.log.error(f"MS file {msdir}/{msname} does not exist. Please check that is where it should be.")
                 raise IOError
+            
+            #if pipeline.enable_task(config, 'fixcalcoords'):
+            tol = config["tol"]
+            tol_diff = config["tol_diff"]
+            #Convert tolerance from arcseconds to radians:
+            tol = tol*np.pi/(180.0*3600.0)
+            tol_diff = tol_diff*np.pi/(180.0*3600.0)
+            db = mkct.calibrator_database()
+            dbc = mkct.casa_calibrator_database()
+            msdict = pipeline.get_msinfo(msname)
+            ra_corr = None
+            dec_corr = None
+            for f in pipeline.bpcal[i]:
+                if field_name != 'target':
+                   if getfield_coords(msdict, f, db, tol = tol, tol_diff = tol_diff):
+                      fielddb, ra_corr, dec_corr = getfield_coords(msdict, f, db, tol = tol, tol_diff = tol_diff)
+                   if getfield_coords(msdict, f, dbc, tol = tol, tol_diff = tol_diff):
+                      fielddb, ra_corr, dec_corr = getfield_coords(msdict, f, dbc, tol = tol, tol_diff = tol_diff)
+                if ra_corr:
+
+                    caracal.log.info("The coordinates of calibrator {0:s} in the MS are offset. This is a known problem for some vintage MeerKAT MSs.".format(f))
```

Comment:
Suggest escalating this into an log.error message (but not an exception) if ``fixcalcoords`` is not enabled, and adding a message to the effect of "you may enable fixcalcoords to attempt to fix this problem automatically". If it is enabled, then issue the messages at info level as they are now.

Reply:
Sounds good.

Reply:
Done.